### ISSUE TITLE
ssh_main returns without printing if no ssh key is found

### ISSUE THREAD

**Issue by IMcD23** [holzschu/ios_system#24]
I've stepped through the ssh_main method to see why nothing is happening when I run `ssh Ian@<ip> pwd`.

I have no SSH keys on my device, and no `.ssh` folder. 

After checking the userauthlist, `rc` = 0. It searches through my `~/Documents` folder in a while loop, and `rc` is still 0. On line 537, it checks whether `rc` is greater than or equal to zero, and returns if so, without printing any error code.

P.S. "Authentification" is not a word, "Authentication" is.

**Comment by holzschu** [comment]
Ah, I see. If the directory does not contain any public key/private key pair, `libssh2_userauth_publickey_fromfile_ex` is never called, and `rc` remains constant. 
Initializing `rc = -1;` on line 492 should produce an error message (not a very useful one, but an error message). 

But with an interactive shell, it might be good to add the opportunity to ask for a password.

**Comment by IMcD23** [comment]
Sounds like a reasonable fix. Did you write this method yourself? Password auth would be 💯.

Also, if no command is entered, you should use something like this to spawn a shell:
```
	/* Request a terminal with 'vt100' terminal emulation */
	rc = libssh2_channel_request_pty(channel, "vt100");
	if (rc) {
		fprintf(stderr, "Failed requesting pty\n");
		return (EXIT_FAILURE);
	}

	/* Open a SHELL on that pty */
	rc = libssh2_channel_shell(channel);
	if (rc) {
		fprintf(stderr, "Unable to request shell on allocated pty\n");
		return (EXIT_FAILURE);
	}
```

**Comment by IMcD23** [comment]
Additionally, you're missing a `fflush` call when stderr is received over SSH, similar to the one that you have for stdout.

**Comment by holzschu** [comment]
Fixed + shell connection over ssh in d6f1d10 
Password auth left for a future issue.
